### PR TITLE
Update default year directive to allow optional month

### PR DESCRIPTION
**Pull request by LukasGasior1**
This adds possibility to specify default month in `Y` directive.
Example usage:

```
Y2016/07

01 a transaction
  ...
```

Transaction date here is 2016/07/01.

Related issue: #362

### CODE REVIEW

File: hledger-lib/Hledger/Data/Types.hs [hunk 1]
```diff
@@ -237,6 +239,7 @@ type Year = Integer
 data Journal = Journal {
   -- parsing-related data
    jparsedefaultyear      :: (Maybe Year)                          -- ^ the current default year, specified by the most recent Y directive (or current date)
+  ,jparsedefaultmonth     :: (Maybe Month)                         -- ^ the current default month, specified by the most recent Y directive containing month part (or current date)
```
Comment (simonmichael) [inline hledger-lib/Hledger/Data/Types.hs]: Could we disallow having a default month without a default year, perhaps jparsedefaultyear :: Maybe (Year, Maybe Month) ?

File: hledger-lib/Hledger/Read/Common.hs [hunk 2]
```diff
@@ -191,14 +198,16 @@ datep = do
...
-  [y,m,d] <- case (dateparts,currentyear) of
-              ([m,d],Just y)  -> return [show y,m,d]
-              ([_,_],Nothing) -> fail $ "partial date "++datestr++" found, but the current year is unknown"
-              ([y,m,d],_)     -> return [y,m,d]
-              _               -> fail $ "bad date: " ++ datestr
+  currentmonth <- getMonth
+  [y,m,d] <- case (dateparts,currentyear,currentmonth) of
+              ([d],Just y,Just m) -> return [show y,show m,d]
+              ([m,d],Just y,_)    -> return [show y,m,d]
+              ([_,_],Nothing,_)   -> fail $ "partial date "++datestr++" found, but the current year is unknown"
```
Comment (simonmichael) [inline hledger-lib/Hledger/Read/Common.hs]: Could you add an error message for the case when the month is unknown

**Comment by simonmichael** [comment]
Thanks! Comments added to the commit, could you also update the journal manual (hledger-lib/doc/hledger_journal.5.m4.md).

**Comment by LukasGasior1** [comment]
@simonmichael Thanks for your comments, I've updated the code accordingly

**Comment by simonmichael** [comment]
Hmm, I don't remember why I didn't merge this. I think the mail list was unenthusiastic and there didn't seem to be quite enough value to justify another Ledger incompatibility ?

**Comment by LukasGasior1** [comment]
I think that was the case

**Pull request closed without merging**